pkg/api/v1alpha1/cloudstackmachineconfig_types.go

```diff
@@ -0,0 +1,160 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// CloudStackMachineConfigSpec defines the desired state of CloudStackMachineConfig
+type CloudStackMachineConfigSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Template refers to a VM image template which has been previously registered in CloudStack. It can either be specified as a UUID or name
+	Template string `json:"template"`
+	// ComputeOffering refers to a compute offering which has been previously registered in CloudStack. It represents a VM’s instance size including number of CPU’s, memory, and CPU speed. It can either be specified as a UUID or name
+	ComputeOffering string `json:"computeOffering"`
+	// DiskOffering refers to a disk offering which has been previously registered in CloudStack. It represents a VM’s disk size and IOPS (Quality of Service) for primary data storage
+	DiskOffering string `json:"diskOffering,omitempty"`
+	// OSFamily is an EKS-A required field which represents the family of the guest OS image. Currently only “redhat“ is supported for CloudStack, which is also the default
+	OSFamily OSFamily `json:"osFamily,omitempty"`
+	// Details represents an arbitrary mapping of strings to strings, which are passed through to the CloudStack VM. Some example details which are commonly set on the VM include cpuOvercommitRatio, memoryOvercommitRatio, Message.ReservedCapacityFreed.Flag, and rootdisksize
+	Details map[string]string `json:"details,omitempty"`
+	// Users utilizes cloud-init to configure ssh users which will be allowed to log in to the VM using private-key authentication. It consists of an array of objects containing the user name, as well as a list of their authorized public keys
```

Comment:
super nit: I would advise not use the term cloud-init here, since some OS's might not support cloud-init and could need a different format.

Reply:
is "os-specific startup scripts" a better way to capture that?

Reply:
Would it be enough to mention that these will be the users and keys authorized to ssh into the machines?